runtime/gc_base/ContinuationObjectList.hpp

```diff
@@ -0,0 +1,152 @@
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef CONTINUATIONOBJECTLIST_HPP_
+#define CONTINUATIONOBJECTLIST_HPP_
+
+#include "j9.h"
+#include "j9cfg.h"
+#include "modron.h"
+
+#include "BaseNonVirtual.hpp"
+#include "ObjectAccessBarrier.hpp"
+
+class MM_EnvironmentBase;
+
+/**
+ * A global list of Continuation objects.
+ */
+class MM_ContinuationObjectList : public MM_BaseNonVirtual
+{
+/* data members */
+private:
+	volatile j9object_t _head; /**< the head of the linked list of Continuation objects */
+	j9object_t _priorHead; /**< the head of the linked list before Continuation object processing */
+	MM_ContinuationObjectList *_nextList; /**< a pointer to the next Continuation list in the global linked list of lists */
+	MM_ContinuationObjectList *_previousList; /**< a pointer to the previous Continuation list in the global linked list of lists */
+#if defined(J9VM_GC_VLHGC)
+	uintptr_t _objectCount; /**< the number of objects in the list */
+#endif /* defined(J9VM_GC_VLHGC) */
+protected:
+public:
+
+/* function members */
+private:
+protected:
+public:
+	/**
+	 * Add the specified linked list of objects to the buffer.
+	 * The objects are expected to be in a NULL terminated linked
+	 * list, starting from head and end at tail.
+	 * This call is thread safe.
+	 *
+	 * @param env[in] the current thread
+	 * @param head[in] the first object in the list to add
+	 * @param tail[in] the last object in the list to add
+	 */
+	void addAll(MM_EnvironmentBase* env, j9object_t head, j9object_t tail);
+
+	/**
+	 * Fetch the head of the linked list, as it appeared at the beginning of Continuation object processing.
+	 * @return the head object, or NULL if the list is empty
+	 */
+	MMINLINE j9object_t getHeadOfList() { return _head; }
+
+	/**
+	 * Move the list to the prior list and reset the current list to empty.
+	 * The prior list may be examined with wasEmpty() and getPriorList().
+	 */
+	void startContinuationProcessing() {
```

Comment:
'Continuation' is redundant in the name of this method. 

(I know it's the same for Unfinalized and other similar list, but let's not spread it anymore, and sometime we can fix it for others, too).